pkg/kubectl/cmd/get/get.go

```diff
@@ -289,6 +289,86 @@ func (o *GetOptions) Validate(cmd *cobra.Command) error {
 	return nil
 }
 
+type OriginalPositioner interface {
+	OriginalPosition(int) int
+}
+
+type NopPositioner struct{}
+
+func (t *NopPositioner) OriginalPosition(ix int) int {
+	return ix
+}
+
+type RuntimeSorter struct {
+	field      string
+	decoder    runtime.Decoder
+	objects    []runtime.Object
+	positioner OriginalPositioner
+}
+
+func (r *RuntimeSorter) Sort() error {
+	if len(r.objects) == 0 {
+		return nil
+	}
+
+	includesTable := false
+	includesRuntimeObjs := false
+
+	for _, obj := range r.objects {
+		switch t := obj.(type) {
+		case *metav1beta1.Table:
+			includesTable = true
+
+			if err := kubectl.NewTableSorter(t, r.field).Sort(); err != nil {
+				continue
+			}
+		default:
+			includesRuntimeObjs = true
+		}
+	}
+
+	r.positioner = &NopPositioner{}
+
+	if includesRuntimeObjs && includesTable {
+		return fmt.Errorf("sorting is not supported on mixed Table and non-Table object lists")
```

Comment:
@soltysh I attempted to make a mixed list of Table + non-Table resources work. Since we flatten resources, when non-server-printing, we end up with an overall list of resources not grouped by resource kind. This leads to a final sorted output that looks like this:

```
$ kube get all --sort-by .metadata.name --server-print=false
NAME                                                 AGE
deploymentconfig.apps.openshift.io/docker-registry   6d

NAME                      TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
service/docker-registry   ClusterIP   172.30.1.1   <none>        5000/TCP   6d

NAME                                      DESIRED   CURRENT   READY     AGE
replicationcontroller/docker-registry-1   1         1         1         6d
...
```

Since table contents cannot be "flattened", we still end up with n tables, each containing pods, deployments, etc.
Using the TableSorter introduced in this PR, we end up with sorted rows for each table, with output that looks like this:

```
$ kube get all --sort-by .metadata.name
NAME                                READY     STATUS      RESTARTS   AGE
pod/docker-registry-1-s8zm2         1/1       Running     1          6d
pod/persistent-volume-setup-99pzk   0/1       Completed   0          6d
pod/router-1-cbh6r                  1/1       Running     1          6d

NAME                                      DESIRED   CURRENT   READY     AGE
replicationcontroller/docker-registry-1   1         1         1         6d
replicationcontroller/router-1            1         1         1         6d
...
```

Since non-server print sorting depends on "original positions" in order to map sorted objects to their "infos", we lose this data when we attempt to separate Table and non-Table into two lists in order to sort them using their respective sorters.